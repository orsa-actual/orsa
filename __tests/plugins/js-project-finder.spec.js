const fs = require('fs');
const glob = require('glob');
const jsProjectFinder = require('../../src/plugins/js-project-finder');

jest.mock('fs');
jest.mock('glob');

describe('Javascript dependencies', () => {
  it('should scan dependencies', async () => {
    glob.sync.mockReturnValue([]);
    fs.readdirSync.mockReturnValue(['bar']);
    fs.existsSync.mockImplementation((name) => name.indexOf('lerna') < 0);
    fs.readFileSync.mockReturnValue(JSON.stringify({}));
    let obj = null;
    await jsProjectFinder({
      scanDirectories: [
        '.',
      ],
    }, {
      basePath: '/',
      store: {
        update: (inputObj) => {
          obj = inputObj;
        },
      },
      logger: {
        log: () => {},
        error: () => {},
      },
    });
    expect(obj).toEqual({
      id: '311128a3ab878d27e98b4f68914aa64e',
      name: undefined,
      nodeType: 'Project',
      packageJSON: '{}',
      projectType: 'node',
      transient: {
        packageJSON: {},
        path: '/bar',
      },
      version: undefined,
    });
  });

  it('should find lerna projects', async () => {
    glob.sync.mockReturnValue(['zap']);
    fs.readdirSync.mockReturnValue(['bar']);
    fs.existsSync.mockReturnValue(true);
    fs.readFileSync.mockReturnValue(JSON.stringify({}));
    const objs = {};
    await jsProjectFinder({
      scanDirectories: [
        '.',
      ],
    }, {
      basePath: '/',
      store: {
        update: (inputObj) => {
          objs[inputObj.id] = inputObj;
        },
      },
      logger: {
        log: () => {},
        error: () => {},
      },
    });
    expect(objs).toMatchSnapshot();
  });
});
