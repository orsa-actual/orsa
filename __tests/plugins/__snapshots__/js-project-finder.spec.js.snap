// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Javascript dependencies should find lerna projects 1`] = `
Object {
  "311128a3ab878d27e98b4f68914aa64e": Object {
    "id": "311128a3ab878d27e98b4f68914aa64e",
    "name": undefined,
    "nodeType": "Project",
    "packageJSON": "{}",
    "projectType": "node",
    "transient": Object {
      "packageJSON": Object {},
      "path": "/bar/zap",
    },
    "version": undefined,
  },
}
`;
